docs/api-bridging/spec.md

```diff
@@ -105,93 +101,209 @@ However, cross-machine routing is an internal concern of the Desktop Agent Bridg
+
+The `connectedAgentsUpdate` message will take the form:
+
+```typescript
+{
+    type:  "connectedAgentsUpdate",
+    /** Request body, containing the arguments to the function called.*/
+    payload: {
+        /** Should be set when an agent first connects to the bridge and provide its assigned name. */
+        addAgent?: string,
+        /** Should be set when an agent disconnects from the bridge and provide the name that no longer is assigned. */
+        removeAgent?: string,
+        /** Desktop Agent Bridge implementation metadata of all connected agents. 
+         *  Note that this object is extended to include a `desktopAgent` field with the name assigned by the DAB. */
+        allAgents: ImplementationMetadata[],
+        /** The updated state of channels that should be adopted by the agents. SHOULD only be set when an agent is connecting to the bridge. */
+        channelsState?: ChannelState[] // see step4
+    },
+    meta: {
+        /** For a new connection, should be the same as the handshake requestGuid. 
+         *  Should be the same as the responseGuid for a disconnection.
+        */
+        requestGuid: string,
+        /** Unique guid for this message */
+        responseGuid: string,
+        /** Timestamp at which response was generated */
+        timestamp:  date,
+    }
+}
+```
+
+When applying the updated channel state, it should be noted that desktop agents will not have context listeners for previously unknown channels and can simply record that channel's state for use when that channel is first used. For known channel names, the Desktop Agent must also compare its current state to that which it has just received and may need to broadcast context to existing connected listeners. As context listeners can be registered for either a specific type or all types some care is necessary when doing so (as the most recently transmitted Context should be received by un-typed listeners). Hence, updating listeners for a known channel should be performed as follows:
+
+1. The incoming channel state `channelState` for a particular channel should be processed from last (oldest) to first.
+2. If there is no current context of that type, broadcast it to listeners of that specific type only.
+3. If there is a current context of that type, and it does not match the incoming object exactly, broadcast it to listeners of that specific type only.
+4. If the most recent (first in the incoming array) type OR value of that type doesn't match the most recent context broadcast on the channel, broadcast it to un-typed listeners only.
+
+After applying the `connectedAgentsUpdate` message, the newly connected desktop agent and other already connected agents are able to begin communicating through the bridge.
+

```

Comment:
```suggestion

#### Step 6. Disconnects

Although not part of the connection protocol, it should be noted that the `connectedAgentsUpdate` message sent in step 5 should also be sent whenever an agent disconnects from the bridge to update other agents. If any agents remain connected, then the `channelState` does not change and can be omitted. However, if the last agent disconnects the bridge SHOULD discard its internal `channelState`, instead of issuing the update.
```